SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -52,198 +70,251 @@ void ADynamicSubobjectsTest::PrepareTest()
+			if (AssertIsValid(PlayerCharacter, TEXT("PlayerCharacter should be valid"))
+				&& PlayerCharacter == GetFlowPlayerController()->AcknowledgedPawn)
+			{
 				FinishStep();
 			}
-		});
-	}
+		},
+		StepTimeLimit);
 
-	{ // Step 1 - All workers check if they have one AReplicatedGASTestActor in the world, and set a reference to it.
-		AddStep(
-			TEXT("DynamicSubobjectsTestAllWorkers"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
-			[this](float DeltaTime) {
-				TArray<AActor*> FoundActors;
-				UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedGASTestActor::StaticClass(), FoundActors);
+	// Step 3 - The client checks it has the right initial amount of components
+	AddStep(
+		TEXT("DynamicSubobjectsTestClientCheckNumComponents"), FWorkerDefinition::Client(1), nullptr,
+		[this]() {
+			AssertEqual_Int(GetNumComponentsOnTestActor(), 3, TEXT("For this test, DynamicSubobjectTestActor should have 3 components"));
+			FinishStep();
+		},
+		nullptr);
 
-				if (FoundActors.Num() == 1)
-				{
-					TestActor = Cast<AReplicatedGASTestActor>(FoundActors[0]);
+	// Step 4 - The server adds the new dynamic component
+	AddStep(
+		TEXT("DynamicSubobjectsTestServerAddComponent"), FWorkerDefinition::Server(1), nullptr,
+		[this]() {
+			// server has 1 more component than client (idk why)
+			AssertEqual_Int(GetNumComponentsOnTestActor(), 3, TEXT("For this test, DynamicSubobjectTestActor should have 3 components"));
 
-					if (IsValid(TestActor))
-					{
-						FinishStep();
-					}
-				}
-			},
-			StepTimeLimit);
-	}
+			// add new dynamic component to test actor
+			UStaticMeshComponent* AddedComponent = NewObject<UStaticMeshComponent>(TestActor, TEXT("ToRemoveComponent"));
+			AddedComponent->AttachToComponent(TestActor->GetRootComponent(), FAttachmentTransformRules::KeepWorldTransform);
+			AddedComponent->RegisterComponent();
+			AddedComponent->SetIsReplicated(true);
+
+			AssertEqual_Int(GetNumComponentsOnTestActor(), 4, TEXT("Now DynamicSubobjectTestActor should have 4 components"));
+			FinishStep();
+		},
+		nullptr);
+
+	// Step 5 - The client checks it sees the new component
+	AddStep(
+		TEXT("DynamicSubobjectsTestClientSeeNewComponent"), FWorkerDefinition::Client(1), nullptr, nullptr,
+		[this](float DeltaTime) {
+			RequireEqual_Int(GetNumComponentsOnTestActor(), 4, TEXT("Now DynamicSubobjectTestActor should have 4 components"));
+			FinishStep();
+		},
+		StepTimeLimit);
+
+	for (int i = 0; i < DynamicComponentsPerClass + 2; ++i)
+	{
+		const bool LastStepLoop = i == DynamicComponentsPerClass + 1;
+
+		// Step 6 - Server moves the TestMovementCharacter of Client 1 to a remote location, so that it does not see the
+		// AReplicatedGASTestActor.
+		AddStep(TEXT("DynamicSubobjectsTestServerMoveClient1"), FWorkerDefinition::Server(1), nullptr, [this]() {
+			if (ClientOneSpawnedPawn->SetActorLocation(CharacterRemoteLocation)
+				&& RequireEqual_Vector(ClientOneSpawnedPawn->GetActorLocation(), CharacterRemoteLocation,
```

Comment:
this is in a one-shot step, I think I would prefer just
```
move()
assertequal_vector()
finishstep()
```

Reply:
done